spec/haproxy/templates/haproxy_config/global_and_default_options_spec.rb

```diff
@@ -491,4 +491,16 @@
       expect(defaults).to include('option allbackups')
     end
   end
+
+  context 'when ha_proxy.always_allow_body_http10 is true' do
+    let(:properties) do
+      {
+        'always_allow_body_http10' => true
+      }
+    end
+
+    it 'sets the global option' do
+      expect(global).to include('h1-accept-payload-with-any-method')
+    end
+  end
```

Comment:
I would expect tests more akin to [the regex fix, where we use actual requests and verify the behaviour / response codes](https://github.com/cloudfoundry/haproxy-boshrelease/pull/276/files#diff-9edf578784347fdf4f72aa066f00020fe3f84d2f04d0e33aa2c1801fff961372)

Reply:
right... ok.

for reference:
```go
        // The protocol version for incoming server requests.
	//
	// For client requests, these fields are ignored. The HTTP
	// client code always uses either HTTP/1.1 or HTTP/2.
	// See the docs on Transport for details.
	Proto      [string](https://pkg.go.dev/builtin#string) // "HTTP/1.0"
	ProtoMajor [int](https://pkg.go.dev/builtin#int)    // 1
	ProtoMinor [int](https://pkg.go.dev/builtin#int)    // 0
```